src/components/Proposal/ProposalData.tsx

```diff
@@ -79,10 +80,14 @@ class ProposalData extends React.Component<IProps, IState> {
     // Don't schedule timeout if its too long to wait, because browser will fail and trigger the timeout immediately
     const millisecondsUntilExpires = closingTime(this.props.data[0]).diff(moment());
     if (!this.state.expired && millisecondsUntilExpires < 2147483647) {
-      setTimeout(() => { this.setState({ expired: true });}, millisecondsUntilExpires);
+      this.expireTimeout = setTimeout(() => { this.setState({ expired: true });}, millisecondsUntilExpires);
     }
```

Comment:
I don't see a reasons for this. 1) `setState` is meant to be called from `componentDidMount`,  2) the console msg indicates that the component has not yet mounted, which would not be the case in `componentDidMount`.

Reply:
The console message suggests this is happening in `WithSubscriptionComponent`, which calls `setState` in a lot of places.

Reply:
The console message suggests this is happening in `WithSubscriptionComponent`, which calls `setState` in a lot of places.

Reply:
that error message usually occurs when the component is unmounted, and it usually indicates that something setup in componentDidMount was not removed/unset/cleared in componentWillUnmount. So in this case for example the timeout expires/fires and the component is no longer mounted so then this error fires. Thus this fix.

Reply:
sorry, misread the code, was thinking that you added a call to setTimeout that wasn't there before, but I see that is not the case.